### DISCUSSION TITLE
Language server do not shut down even if I close neovim

### DISCUSSION THREAD

**Discussion by Ginkooo** [neoclide/coc.nvim#4732] [Q&A] [answered]
Hey,

I just wanted to ask. I program in Python and Rust using neovim with coc-vim and coc-rust-analyzer and coc-pyright enabled.

I've installed them using `:CocInstall` command

I see in htop the language servers don't shutdown even if I close neovim, even after few hours from last time I'm done using it. Is this a normal behaviour?

If this the right place to ask that question? I could provide more logs if they are needed. Thats my .vimrc: https://github.com/Ginkooo/configs/blob/main/.vim/.vimrc

In CocConfig i just have these two lines:

```json
{
    "python.formatting.provider": "black",
    "coc.preferences.formatOnSave": true
}
```

The processes copied from `ps aux` are:

```
ginkooo   859037  0.2  0.7 22008564 92944 ?      Ssl  19:01   0:01 /usr/bin/node --no-warnings /home/ginkooo/.local/share/nvim/plugged/coc.nvim/build/index.js
ginkooo   277209  0.0  1.6 1128316 198804 ?      Sl   Aug24   0:09 /usr/bin/node --max-old-space-size=3072 /home/ginkooo/.config/coc/extensions/node_modules/coc-pyright/node_modules/pyright/langserver.index.js --
ginkooo   327643  0.0 10.4 2331492 1262984 ?     Sl   Aug24   1:13 /home/ginkooo/.config/coc/extensions/coc-rust-analyzer-data/rust-analyzer
```

They consume quite a lot of memory.

Thanks :)

PS

```
NVIM v0.9.1
Build type: Release
LuaJIT 2.1.1692616192
```

**Comment by fannheyward** [comment]
> language servers don't shutdown even if I close neovim

Can't reproduce, after you close nvim, coc.nvim client and all servers that launched by coc.nvim will be terminated. How do you close nvim?

**Reply by Ginkooo** [reply]
Hay, sorry for the late response. It happens both when I close by ":q" and closing the whole terminal window. Now following the below response I added autocmd to close it to .vimrc but It does not close on terminal window close

**Comment by chemzqm** [comment, marked as answer]
Seems coc.nvim process not killed as expected, checkout `:h g:coc_process_pid`

**Reply by Ginkooo** [reply]
Wow, it works! When i close using :q and :q! it closes all the coc processes correctly :) Thanks!

But is there a way it will close when I close the terminal window using my window manager too? I use i3 and often I just use a keyboard shortcut to close the vim window

**Reply by chemzqm** [reply]
Should work if `VimLeavePre` autocmd triggered.
